server/pulp/plugins/util/misc.py

```diff
@@ -142,3 +143,89 @@ def clear_directory(path, skip_list=()):
+
+
+def make_link_unit(unit, extra_dst_path, working_dir,
+                   remote_repo_path, remote_root, configuration, extra_src_path, out_path):
+    """
+    This method creates symlink in working directory, pointing to remote
+    content directory, so they can be sync to remote server
+
+    :param repo_type: type of repository
+    :type repo_type: str
+    :param unit: Pulp content unit
+    :type unit: Unit
+    :param extra_dst_path: ending path for processes units
+                          (for example 'Packages' for rpm)
+    :type extra_dst_path: str
+    :param working_dir: Working directory
+    :type working_dir: str
+    :param remote_repo_path: relative repo destination path on remote server
+    :type remote_repo_path: str
+    :param remote_root: remote destination root directory
+    :type remote_root: str
+    :param configuration: configuration of distributor
+    :type configuration: dict
+    """
+    filename = os.path.basename(unit.storage_path)
+
+    if not os.path.exists(os.path.join(working_dir, extra_dst_path, *extra_src_path)):
+        os.makedirs(os.path.join(working_dir, extra_dst_path, *extra_src_path))
+
+    origin_path = get_origin_rel_path(unit, configuration)
+
+    dest = os.path.normpath(os.path.join(working_dir,
+                                         *([extra_dst_path] + extra_src_path + [filename])))
+    link_source = os.path.relpath(os.path.normpath(os.path.join(remote_root,
+                                                                os.path.dirname(origin_path))),
+                                  os.path.normpath(os.path.join(remote_root,
+                                                                remote_repo_path.lstrip("/"),
+                                                                out_path, extra_dst_path)))
+
+    _log.info("LN %s -> %s " % (link_source, dest))
+    if os.path.islink(dest):
+        os.remove(dest)
+    os.symlink(link_source, dest)
+    return os.path.join(*([extra_dst_path] + extra_src_path + [filename]))
+
+
+def get_origin_rel_path(unit, config):
+    """
+    return relative path of content unit in remote fs
+
+    :param unit: Pulp content unit
+    :type unit: Unit
+    :param configuration: configuration of distributor
+    :type configuration: dict
+    """
+
+    storage_dir = pulp_config.get('server', 'storage_dir')
+    content_base = os.path.join(storage_dir, "content", unit.metadata["_content_type_id"])
+    rel_unit_path = os.path.relpath(unit.storage_path, content_base)
+    remote_content_base = get_remote_content_base(config)
+    remote_content_type_dir = unit.metadata["_content_type_id"]
+    return os.path.join(remote_content_base, remote_content_type_dir, rel_unit_path)
+
+
+def get_remote_content_base(config):
+    remote_content_base = config.get("relative_url")
+    if not remote_content_base:
+        remote_content_base = os.path.join("content")
+    return remote_content_base
+
+
+def strbool2bool(_str):
```

Comment:
I'm hoping we can remove this. It looks like each use of it could or should not have any questions about what type it's dealing with.